purchases/src/defaults/kotlin/com/revenuecat/purchases/Purchases.kt

```diff
@@ -839,6 +840,61 @@ class Purchases internal constructor(
             purchasesOrchestrator.allowSharingPlayStoreAccount = value
         }
 
+    /**
+     * The preferred UI locale override for RevenueCat UI components.
+     * This affects both API requests and UI rendering.
+     *
+     * @return The preferred UI locale override, or null if using system default
+     */
+    val preferredUILocaleOverride: String?
+        @Synchronized get() = purchasesOrchestrator.preferredUILocaleOverride
+
+    /**
+     * Override the preferred UI locale for RevenueCat UI components at runtime.
+     * This affects both API requests and UI rendering.
+     *
+     * If the locale changes, this will automatically clear the offerings cache and trigger
+     * a background refetch to get paywall templates with the correct localizations.
+     *
+     * @param localeString The locale string (e.g., "es-ES", "en-US") or null to use system default
+     * @return true if cache was cleared and refetch triggered, false if locale unchanged or cache clear rate limited
+     */
+    fun overridePreferredUILocale(localeString: String?): Boolean {
+        val previousLocale = purchasesOrchestrator.preferredUILocaleOverride
+        debugLog { "overridePreferredUILocale: changing from '$previousLocale' to '$localeString'" }
+        purchasesOrchestrator.preferredUILocaleOverride = localeString
+
+        return if (previousLocale != localeString) {
+            debugLog { "Locale changed, attempting to fetch fresh offerings" }
+            val fetchTriggered = fetchOfferingsIfPossible()
+            debugLog { "Fresh fetch result: $fetchTriggered" }
+            fetchTriggered
+        } else {
+            debugLog { "Locale unchanged, no fresh fetch needed" }
+            false // Locale didn't change, no fresh fetch needed
+        }
+    }
+
+    /**
+     * Fetches fresh offerings from the API to get updated localizations.
+     * This is useful after changing the preferred locale to get paywall templates
+     * with the correct localizations.
+     *
+     * This method is rate limited to 2 calls per 60 seconds to prevent excessive
+     * network requests.
+     *
+     * @return true if fresh fetch was triggered, false if rate limited
+     */
+    private fun fetchOfferingsIfPossible(): Boolean {
```

Comment:
Hmm I would remove this method and call the purchasesOrchestrator directly and add the logs there if needed. The least logic we have in the facade the better.